invertedai/api/validations.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Optional, Dict
+from invertedai.error import InvalidInputType, InvalidInput
+from invertedai.common import (
+    RecurrentState,
+    AgentState,
+    AgentAttributes,
+    TrafficLightId,
+    TrafficLightState,
+)
+
+
+def drive_validation(drive):
+    def wrapped_drive(
```

Comment:
This is unnecessarily complicated. There is no reuse, so you can just insert the validation code block into the relevant function (drive/initialize/etc.).